Does not work with GHC 9.0.2

<details>
<summary><code>cabal build --with-compiler ghc-9.0.2</code></summary>

```
% cabal build --with-compiler ghc-9.0.2
Build profile: -w ghc-9.0.2 -O1
In order, the following will be built (use -v for more details):
 - heterocephalus-1.0.5.4 (lib) (first run)
Preprocessing library for heterocephalus-1.0.5.4..
Building library for heterocephalus-1.0.5.4..
[6 of 6] Compiling Text.Heterocephalus ( src/Text/Heterocephalus.hs, /tmp/heterocephalus-1.0.5.4/dist-newstyle/build/x86_64-linux/ghc-9.0.2/heterocephalus-1.0.5.4/build/Text/Heterocephalus.o, /tmp/heterocephalus-1.0.5.4/dist-newstyle/build/x86_64-linux/ghc-9.0.2/heterocephalus-1.0.5.4/build/Text/Heterocephalus.dyn_o )

src/Text/Heterocephalus.hs:498:10: error:
    • Couldn't match type ‘[Stmt] -> Exp’ with ‘Exp’
      Expected: Q Exp
        Actual: Q ([Stmt] -> Exp)
    • In the expression: return $ DoE $ map NoBindS exps
      In a case alternative: _ -> return $ DoE $ map NoBindS exps
      In a stmt of a 'do' block:
        case exps of
          [] -> [| return () |]
          [x] -> return x
          _ -> return $ DoE $ map NoBindS exps
    |
498 |     _ -> return $ DoE $ map NoBindS exps
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Text/Heterocephalus.hs:498:25: error:
    • Couldn't match expected type: Maybe
                                      Language.Haskell.TH.Syntax.ModName
                  with actual type: [Stmt]
    • In the second argument of ‘($)’, namely ‘map NoBindS exps’
      In the second argument of ‘($)’, namely ‘DoE $ map NoBindS exps’
      In the expression: return $ DoE $ map NoBindS exps
    |
498 |     _ -> return $ DoE $ map NoBindS exps
    |                         ^^^^^^^^^^^^^^^^
```

</details>

Seems to be a Template Haskell issue. I am working on a patch.

--- a/heterocephalus.cabal
+++ b/heterocephalus.cabal
@@ -43,6 +43,7 @@ library
                      , parsec >= 3.1
                      , shakespeare >= 2.0
                      , template-haskell >= 2.12
+                     , template-haskell-compat-v0208
                      , text >= 1.2
                      , transformers >= 0.5
   ghc-options:         -Wall -Wcompat -Wnoncanonical-monad-instances

--- a/src/Text/Heterocephalus.hs
+++ b/src/Text/Heterocephalus.hs
@@ -79,6 +79,7 @@ import Language.Haskell.TH.Syntax
 #else
 import Language.Haskell.TH.Syntax
 #endif
+import TemplateHaskell.Compat.V0208
 import Text.Blaze (preEscapedToMarkup)
 import Text.Blaze.Html (toHtml)
 import Text.Blaze.Internal (preEscapedText)
@@ -495,7 +496,7 @@ docsToExp set scope docs = do
   case exps of
     [] -> [|return ()|]
     [x] -> return x
-    _ -> return $ DoE $ map NoBindS exps
+    _ -> return $ doE $ map NoBindS exps
 
 docToExp :: HeterocephalusSetting -> Scope -> Doc -> Q Exp
 docToExp set scope (DocForall list idents inside) = do
@@ -525,11 +526,7 @@ docToExp set scope (DocCond conds final) = do
       return $ nonUnaryTupE [d', docs']
       where
         nonUnaryTupE :: [Exp] -> Exp
-        nonUnaryTupE es  = TupE $
-#if MIN_VERSION_template_haskell(2,16,0)
-                                 map Just
-#endif
-                                 es
+        nonUnaryTupE es  = tupE es
 docToExp set scope (DocCase deref cases) = do
     let exp_ = derefToExp scope deref
     matches <- mapM toMatch cases

